collector/infiniband_linux.go

```diff
@@ -108,141 +88,55 @@ func NewInfiniBandCollector() (Collector, error) {
 	return &i, nil
 }
 
-// infinibandDevices retrieves a list of InfiniBand devices.
-func infinibandDevices(infinibandPath string) ([]string, error) {
-	devices, err := filepath.Glob(filepath.Join(infinibandPath, "/*"))
-	if err != nil {
-		return nil, err
-	}
-
-	if len(devices) < 1 {
-		log.Debugf("Unable to detect InfiniBand devices")
-		err = errInfinibandNoDevicesFound
-		return nil, err
-	}
-
-	// Extract just the filenames which equate to the device names.
-	for i, device := range devices {
-		devices[i] = filepath.Base(device)
-	}
-
-	return devices, nil
+func (c *infinibandCollector) pushMetric(ch chan<- prometheus.Metric, name string, value uint64, deviceName string, port string, valueType prometheus.ValueType) {
+	fieldDesc := c.metricDescs[name]
```

Comment:
Unnecessary allocation - just refer to `c.metricDescs[name]` directly in the line below.